### ISSUE TITLE
Update Intervals not working

### ISSUE THREAD

**Issue by Steinbichler** [davidrapan/ha-solarman#84]
First thank you for this fine solarman integration! Great Job!

I am running homeassistant in actual version. My converter is a Deye 600. Relais is installed.

I have your ha-slarman up an running installed via HACS. Nearly everthing works fine but:

I have requested at Deye support to set the update interval for my Deye 600 to 1 Minute. 
They agreed to have it done. I see in the Homepage of the converter the update time is set to 1.

But nothing changed in the ha-solarman. I changed the number in the definition file: deye_2mppt.yaml.
After the installation the update_interval: 5 was set. 
What ever the number I enter, and for sure I did a restart after the change, nothing happens. 
The update interval is always 5 Minutes.

Can you help please?

Thank you an regards Roland

**Comment by davidrapan** [comment]
Hi @Steinbichler, I'm little bit worried that you are mixing different things here together. 

> I have requested at Deye support to set the update interval for my Deye 600 to 1 Minute.
They agreed to have it done. I see in the Homepage of the converter the update time is set to 1.

This integration and HA has nothing to do with Deye and their support and is completely separated thing. No configuration from Deye side won't affect this integration at all.

And at the same time nothing set in this integration will affects Deye Cloud nor Solarman App. Again it is completely separate thing from those.

> But nothing changed in the ha-solarman. I changed the number in the definition file: deye_2mppt.yaml.
After the installation the update_interval: 5 was set.
What ever the number I enter, and for sure I did a restart after the change, nothing happens.
The update interval is always 5 Minutes.

What are you trying to achieve? update_interval: 5 in the 2mppt.yaml means 5 seconds not 5 minutes.

**Comment by Steinbichler** [comment]
Hi if set to 5 as ist should be 5 seconds I have this:
![Bildschirmfoto vom 2024-08-15 16-26-09](https://github.com/user-attachments/assets/009e3b30-50ee-48e9-b29a-1330f7a439dd)
Update was 6 Minutes before.

**Comment by davidrapan** [comment]
What shows Update Interval sensor in diagnostic card in device detail view?

**Comment by CrazyUs3r** [comment]
@Steinbichler 
Wie gut ist das wlan Signal 
Mein m80 macht auch nur alle 5min ist aber auch auf 1min gestellt. In der solarman app kommen die Signale aber auch nur im 5min intervall an

**Comment by Steinbichler** [comment]
@davidrapan
![Bildschirmfoto vom 2024-08-15 19-01-07](https://github.com/user-attachments/assets/1473f471-c2e0-413c-ad50-2fa00bfa24e4)

do you mean this one

@CrazyUs3r 
Das Wlan kann das Problem nicht sein es ist sehr gut und in der Solarman App kommt der update nach einer Minute

**Comment by Steinbichler** [comment]
[home-assistant_solarman_2024-08-15T17-36-23.029Z.log](https://github.com/user-attachments/files/16628192/home-assistant_solarman_2024-08-15T17-36-23.029Z.log)

I hope it helps.Thank you an Regards Roland

**Comment by davidrapan** [comment]
What the actual f*ck is happening in there!

Could I please ask you to do the following please:
1. Disable debug logging (if you have it enabled)
2. Restart Home Assistant
3. Remove the device from the integration
4. Enable debug logging
5. Add your device back again and go trough configuration
6. Let it run for some time
7. Disable debug logging and post here the log file again

Thank you for your cooperation! 😉

**Comment by PV-Joe** [comment]
I remember this behavior after my first installation.

The Update information in seconds or minutes from Home Assistant was showing wrong values but the sensors updates as expected.

I haven’t done anything about it and wondered. The problem desolved by itself after a while.

**Comment by Steinbichler** [comment]
@davidrapan 

Hallo I have done it in your requested way. After the update and the delete of the converter in HO the automatic recognition does not work any more,

![Bildschirmfoto vom 2024-08-16 09-55-43](https://github.com/user-attachments/assets/dc68f0a7-0e69-4bbf-87c2-2b52f1cb0790)

![Bildschirmfoto vom 2024-08-16 09-55-43](https://github.com/user-attachments/assets/74ba5351-9ec6-4bb7-81b5-fd59ec903c0f)

and the Log file

[home-assistant_solarman_2024-08-16T07-55-56.658Z.log](https://github.com/user-attachments/files/16634545/home-assistant_solarman_2024-08-16T07-55-56.658Z.log)

**Comment by Steinbichler** [comment]
Sorry about the confusion in the deleted comment.
It was a problem with my HO installation everything woks fine I will upload the log asap.
Regards Roland

**Comment by CrazyUs3r** [comment]
My log
[home-assistant_solarman_2024-08-16T08-13-04.070Z.log](https://github.com/user-attachments/files/16634730/home-assistant_solarman_2024-08-16T08-13-04.070Z.log)

**Comment by davidrapan** [comment]
@Steinbichler and when you are observing the value does it really update with such a long period? Or just the HA shows it incorrectly?

**Comment by davidrapan** [comment]
Maybe [fix: Proper implementation of sensor platform](https://github.com/davidrapan/ha-solarman/commit/16b4522b51d8006814d84438c7dc42c20e1d7a37) will do something about that strange update behavior.

**Comment by CrazyUs3r** [comment]
I test it also

**Comment by Steinbichler** [comment]
@davidrapan.

I did a test to cover my pannels. And yes ist last at least the 5 to 6 Minutes till I see the change in HA.

A question to my understanding: Even if the content of the register in my inverter did not change,  the update in HA-solarman should read the unchanged content after the interval time? Is it not in this way?

And for the link in your comment to the definition file? Should I change ist like it is in the description? and clear all the names?

**Comment by davidrapan** [comment]
> I did a test to cover my pannels. And yes ist last at least the 5 to 6 Minutes till I see the change in HA.

Fuu, that is really strange :-/

> A question to my understanding: Even if the content of the register in my inverter did not change, the update in HA-solarman should read the unchanged content after the interval time? Is it not in this way?

It will be read from the inverter. BUT! **Ha won't update it's "Last updated" until the value really changes.**

> And for the link in your comment to the definition file? Should I change it like it is in the description? and clear all the names?

**It was link to a commit, it affects whole source code not just definition profile.**

**Comment by CrazyUs3r** [comment]
![Uploading 20240816_134117.jpg…]()

**Comment by davidrapan** [comment]
@CrazyUs3r, you did not wait long enough for the file to finish uploading, hehe.

**Comment by davidrapan** [comment]
@Steinbichler Try [v24.08.17](https://github.com/davidrapan/ha-solarman/releases/tag/v24.08.17) and see if it won't fix your issue.

**Comment by bjoernmayer** [comment]
Heyho,
I am an `24.08.17` and I can see the same behaviour. Data is updated around every 5ish minutes

**Comment by davidrapan** [comment]
Hey @bjoernmayer. Please enable debug logging and post the log here.

**Comment by Steinbichler** [comment]
Hallo I am on 24.08.17 here again a log from my Installation.
[home-assistant_solarman_2024-08-19T14-20-56.963Z.log](https://github.com/user-attachments/files/16661444/home-assistant_solarman_2024-08-19T14-20-56.963Z.log)
Regards

**Comment by davidrapan** [comment]
Hey @Steinbichler, from sensor named "Fault information Word 2" remove:
```
        lookup:
          - key: 0x0000
            value: "OK"
```

**Comment by Steinbichler** [comment]
Hallo davidrapan done, but no change.
Here again a logfile from the time after the change in the devinition file.

[home-assistant_solarman_2024-08-19T16-00-04.808Z.log](https://github.com/user-attachments/files/16662697/home-assistant_solarman_2024-08-19T16-00-04.808Z.log)

![Bildschirmfoto vom 2024-08-19 17-35-39](https://github.com/user-attachments/assets/e28af66e-41d4-4c92-9bcf-a22d16efe83e)
![Bildschirmfoto vom 2024-08-19 17-34-54](https://github.com/user-attachments/assets/4af0542e-4bf2-4409-96fd-a769ad752285)

**Comment by CrazyUs3r** [comment]
i have add a 2mppt.yaml and hier my log
[home-assistant_solarman_2024-08-19T16-00-44.380Z.log](https://github.com/user-attachments/files/16662711/home-assistant_solarman_2024-08-19T16-00-44.380Z.log)

**Comment by davidrapan** [comment]
@Steinbichler, according to the log there is clearly visible that HA talks with the inverter every 5 seconds and also gets the data correctly.

Whatever is causing the issue is happening after that in the HA cause there are also others like here for example @CrazyUs3r with the same device and it's working okay for him... So I'm really out of ideas what's causing issues for you.

Do you have the auto polling enabled?

**Comment by CrazyUs3r** [comment]
By the way, I still have the upload time of 5 minutes. I tried to change it with github.com/s10 his exe but it doesn't work. I already wrote to deye. I have also already changed was @davidrapan changed and everything that was done.

**Comment by CrazyUs3r** [comment]
No i have 1min
![Screenshot_20240819_191836_Deye Cloud](https://github.com/user-attachments/assets/07e42566-18cc-43df-aa4a-aeb1e9a425dc)

**Comment by davidrapan** [comment]
@CrazyUs3r, this is irrelevant for our integration.

**Comment by davidrapan** [comment]
I have also on the cloud refresh rates in minutes but it just means that servers from deye/solarman are asking for data in that interval.

**Comment by CrazyUs3r** [comment]
Update is comming in 1min
![Screenshot_20240819_194408_Home Assistant](https://github.com/user-attachments/assets/fc4274f4-8110-434c-9d48-9dd8cf6e1a1d)

**Comment by davidrapan** [comment]
@CrazyUs3r What about PV1 Power also every 1m?

**Comment by davidrapan** [comment]
I just checked it and it your devices really are not giving out new data with every request... My device does (It's also deye, even with set that on deye cloud to 1m) so i thought it should work the same for you guys too.

**But that means it should be set somewhere in the device, right?**

**Comment by davidrapan** [comment]
@CrazyUs3r, can you show me screen of your: http://{inverter_ip}/config_hide.html

**Comment by CrazyUs3r** [comment]
@davidrapan sure I'll send it to you tomorrow today it's too dark

**Comment by CrazyUs3r** [comment]
@davidrapan here all from ha
![Screenshot_20240819_203046_Home Assistant](https://github.com/user-attachments/assets/b18ebded-783d-4d21-b445-c883c7c8cf59)

**Comment by davidrapan** [comment]
Nono i was talking about http://{inverter_ip}/config_hide.html it's a page for the stick logger.

**Comment by davidrapan** [comment]
Ok.. mh..

**Comment by davidrapan** [comment]
What happens when you switch it to transparency and hit save?

**Comment by CrazyUs3r** [comment]
I haven't figured out what the difference is yet. I switched last year and nothing changed. Maybe you know what the difference is.

**Comment by davidrapan** [comment]
I was wondering if it maybe won't affect the update_interval in some way.

**Comment by CrazyUs3r** [comment]
So now it is separated again and all data is gone
![Screenshot_20240819_203800_Home Assistant](https://github.com/user-attachments/assets/165f8b22-fd16-4ba9-aaf2-7b07f4450a49)

**Comment by davidrapan** [comment]
What shows the log?

**Comment by davidrapan** [comment]
Anyway wrap up of this issue: As the inverter itself is the one who is sending (old data) and changing them for us once every 1m for @CrazyUs3r or 5m for @Steinbichler it's is sadly not an issue of the integration but either some sort of a hidden setting of the inverter cause it looks like this period can be at least change to 5m or 1m or it's hardware limitation. :-/

**Comment by CrazyUs3r** [comment]
> You mean cause of night, right?

Yes

**Comment by davidrapan** [comment]
Would like to know from where that value of 60s or 5m comes from.

**Comment by CrazyUs3r** [comment]
That's a good question in my m80 the solarman says 1min for upload 60s for the other but the values ​​in the app are only in 5min intervals

**Comment by davidrapan** [comment]
And did you also try deye cloud instead of solarman app?

**Comment by CrazyUs3r** [comment]
I just have both apps to look at, but I'm slowly moving over to the deye cloud

**Comment by davidrapan** [comment]
And is it the same? The update_interval?

I'm currently not using any of the apps cause now I'm finally completely satisfied with what my integration + HA is offering to me so I really not just not need them but they don't even remotely offer comparable value at all.

And soon I'm gonna also try to disconnect my devices from their servers completely.

**Comment by CrazyUs3r** [comment]
Yes, both apps show the same thing.

I'm on your side when it comes to your decision to separate. The project in HA is running very well for me too.

Oh, if you want to separate, the update time won't be cut off any more. Or the daily values ​​won't be reset.

**Comment by CrazyUs3r** [comment]
Look in my log and I see that data is pulled every 5s

[home-assistant_solarman_2024-08-20T10-05-02.693Z.log](https://github.com/user-attachments/files/16674272/home-assistant_solarman_2024-08-20T10-05-02.693Z.log)

**Comment by davidrapan** [comment]
Yes they are but the content of the data is the same every time. That's the issue.

**Comment by CrazyUs3r** [comment]
I agree with you completely and always overwriting data with the same value would not be any better

**Comment by bjoernmayer** [comment]
Heyho,

this is my `config_hide.html`:
![grafik](https://github.com/user-attachments/assets/a64cd754-5ab0-43f8-b6c9-7f818990f13c)

The tcp timeout is set to 5 minutes, but this sounds unrelated

**Comment by CrazyUs3r** [comment]
And what inverter typ??

**Comment by davidrapan** [comment]
@Steinbichler or @bjoernmayer don't you want to also try to write to your supp as @CrazyUs3r did to lower that update period as an experiment if it won't (for some "really mysterious" reason) help?

**Comment by bjoernmayer** [comment]
> And what inverter typ??

Deye 600 s3


> @Steinbichler or @bjoernmayer don't you want to also try to write to your supp as @CrazyUs3r did to lower (make it more often) that update period as an experiment if it won't (for some "really mysterious" reason) help?

I just did. Let's see what happens.

**Comment by Steinbichler** [comment]
Hallo David, that is the point I mentioned at the beginning of this item.

I have requested at Deye support to set the update interval for my Deye 600 to 1 Minute.
They agreed to have it done. I see in the Homepage of the converter the update time is set to 1.

I now again send them a mail to ensure they have done it and I got the the answer: Yes we did!

**Comment by davidrapan** [comment]
@Steinbichler. Yeah I know but still it has nothing to do with the integration. Your device is still sending the same data for 5 minutes... What about update interval in their app? Did that change?

**Comment by davidrapan** [comment]
> Here is my log [home-assistant_solarman_2024-08-16T08-23-58.631Z.log](https://github.com/user-attachments/files/16634833/home-assistant_solarman_2024-08-16T08-23-58.631Z.log) ![Bildschirmfoto vom 2024-08-16 10-31-40](https://private-user-images.githubusercontent.com/16692138/358521894-c5123a58-6a80-4a04-859a-17e707eea089.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzMTU0NTAsIm5iZiI6MTcyNDMxNTE1MCwicGF0aCI6Ii8xNjY5MjEzOC8zNTg1MjE4OTQtYzUxMjNhNTgtNmE4MC00YTA0LTg1OWEtMTdlNzA3ZWVhMDg5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIyVDA4MjU1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJhZjE0ZmJmY2NiODFjZmY5MTE4M2FiYWQ2Y2U5ZTA0ZTM4NmZlMmVkNjk5NTE2MjdkNzQzY2ZiNzRiMmM0MDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xu-jr9MUJu_4av11vm-9Gv8mO2usoq7n6xrfnz5IS7Q)

@githubDante? What are those frames starting wit the `aa`?

**Comment by Steinbichler** [comment]
![Bildschirmfoto vom 2024-08-22 12-00-33](https://github.com/user-attachments/assets/50677a8c-cea6-489f-9d9c-0207ab5609eb)

**Comment by CrazyUs3r** [comment]
Same on my

**Comment by CrazyUs3r** [comment]
Last update in this Page i see everytime only 0 and 1... 1 for send 0 for nothing

**Comment by davidrapan** [comment]
Little OT but I would guys suggest you to set Mode Select to: AP/STA from AP+STA. If you are not using the AP of the inverter, ofc.

@CrazyUs3r, interesting. What you guys @Steinbichler and @bjoernmayer have in there?

**Comment by githubDante** [comment]
> @githubDante? What are those frames starting wit the `aa`?

Smart relay or something like that.

**Comment by davidrapan** [comment]
> Smart relay or something like that.

Mh, kay. Thanks!

**Comment by Steinbichler** [comment]
Hallo Done the Change to AP/STA nothing changed.

By the way the sensor "sensor.inverter_update_interval" changes every minute from 0 to 1 and back.

But the shown update time in " sensor.inverter_total_ac_output_power_active" still goes up to 5

**Comment by davidrapan** [comment]
> Hallo Done the Change to AP/STA nothing changed.

That has nothing to do with your problem, it was just a security suggestion.

> By the way the sensor "sensor.inverter_update_interval" changes every minute from 0 to 1 and back.

What are you talking about?

> But the shown update time in " sensor.inverter_total_ac_output_power_active" still goes up to 5

Yes cause the inverter is not giving you new data.

**Comment by CrazyUs3r** [comment]
I just looked at the graphs in the deye cloud app and I see that the values ​​are one minute apart. Just for info

**Comment by terzo33** [comment]
Hi, I have deye 2 micro inverters and i have the same problem. Update interval is about 5 minutes for PV

![Screenshot_20240920_154238_Home Assistant.jpg](https://github.com/user-attachments/assets/225f825f-41d1-45aa-811a-6f422cc7712e)

But about 5 minutes is only for PV information(power, current and voltage), others data updates with 5/6 seconds

![Screenshot_20240920_154413_Home Assistant.jpg](https://github.com/user-attachments/assets/44289128-8807-4573-9efd-498f3261051f)

![Screenshot_20240920_154425_Home Assistant.jpg](https://github.com/user-attachments/assets/ff8f74b3-5f8a-48c2-901b-3e85114fe890)

In solarman and deye cloud apps, data change every 60 seconds

Someone have solution?

Thanks a lot

**Comment by CrazyUs3r** [comment]
I'm starting to think that it's because of ha

**Comment by terzo33** [comment]
> I'm starting to think that it's because of ha

On deye_hybrid all works fine. Interval update is about 5 seconds.

The problem is only on microinverter configuration

**Comment by davidrapan** [comment]
Hi @terzo33 please enable debug logging tomorrow for couple of hours and attach the log file here. I'll analyze it.

**Comment by terzo33** [comment]
> Hi @terzo33 please enable debug logging tomorrow for couple of hours and attach the log file here. I'll analyze it. 

Hi @davidrapan, my log here.

![home-assistant.txt](https://github.com/user-attachments/files/17085671/home-assistant.txt)

I hope we can solve the problem.
Thanks a lot

**Comment by davidrapan** [comment]
Thanks, I'll look into it and get back to u!

**Comment by terzo33** [comment]
@davidrapan can I help you? I'm available for test :)

**Comment by Dedelino** [comment]
I have the same problem described above: "Update Interval" in HA is around 5sec, however "Inverter Power", "PV1 Power" etc. only get updated every 5 minutes. 
I'm using a Deye SUN-M200G4-EU-Q0 with Firmware version MW3_16U_5406_1.59
If there's anything I can do to help, let me know. I really like this integration, this is the only thing that bothers me :(

**Comment by davidrapan** [comment]
The device itself does not provide updated values for PV sensors with lower interval than that so there is nothing we can do in this integration to change that.

I would suggest you to try to ask your manufacturers about it or maybe try to upgrade the firmware? I'm really clueless here...

**Comment by Dedelino** [comment]
I contacted Deye and they offered to set the Cloud update interval to 1 min (instead of 5 min). However, as expected: this didn't change the modbus data update interval (which I mentioned explicitely... i guess they didn't understand the problem :( ) 
May I ask which inverter+firmware you are using?

**Comment by Dedelino** [comment]
I have now tried to explain the problem in various detailed mails to different tech supports. None of them understood the issue and/or the difference between local Modbus connections and the data that is uploaded to the cloud. They all came back to the cloud data upload interval at some point in the discussion.
I think this is a dead end and we‘ll have to see if at some point a Firmware Update will change this behaviour by accident.

**Comment by davidrapan** [comment]
That's unfortunate... I was hoping for at least some explanation.

**Comment by fpfuetsch** [comment]
I can confirm this issue with a DEYE M80G4-EU, firmware Version: MW3_16U_5406_1.59. Sensor like "grid_voltage" update frequently while the "interesting" sensors like "power" update only every 5 minutes.

**Comment by davidrapan** [comment]
Can you guys give me exact list of all sensors which has this long update interval?

**Comment by fpfuetsch** [comment]
From what I can see _grid_* sensors update frequently but everything related to the solar panels *pover, *pv_power, pv1_*, pv2_* is updating exactly every 5 minutes.

**Comment by Adminius** [comment]
I have same problem with my 7x Micro Inverters: 4x SUN1600G3, 2x M160G4, 1xM200G4
![image](https://github.com/user-attachments/assets/fe3f7a07-431d-495f-9154-66a737476594)
Support updated Logger on 3x G4 to 1 min, but still, the data is updated each 5 minutes

**Comment by davidrapan** [comment]
You can check through debug logging that data are fetched evert 5 seconds but microinverter sends the same data (for PV sensors and some other) for duration of 5m (in some cases 1m). So this is not something which can be solved from the integration (or at least I don't know yet how could be device influenced in this matter).

**Comment by Adminius** [comment]
that's weird. I can't really check if cloud's 1 min really 1 min or also 5... hm. 
This makes all my logics fail on cloudy days (EV charging on PV)

**Comment by davidrapan** [comment]
I totally understand! It doesn't make sense at all but I currently do not have any microinverter unit to test things out and maybe find a solution. I do plan to acquire some as soon I'll have some spare funds to "waste" to do exactly that. 😉

**Comment by Adminius** [comment]
Can I somehow help?
Theoreticaly I can provide VPN to one of my inverters

**Comment by davidrapan** [comment]
Not really sure if VPN would be enough as my idea is to sniff out traffic between the device and the internet in MITM style and ask support to reconfigure that update interval and it will hopefully reveal something.

**Comment by Adminius** [comment]
Maybe?
https://github.com/Hypfer/deye-microinverter-cloud-free/tree/master/dummycloud

**Comment by davidrapan** [comment]
That is for disconnecting the device from the cloud and not losing time synchronization in the process (can be done using this integration + HA automation to set time too).

**Comment by davidrapan** [comment]
Can someone try [feat: Implemented response to V5 HEARTBEAT](https://github.com/davidrapan/ha-solarman/commit/1f148857e670ac190ee8a0f52c480974197be665)?

It should solve time synchronization for cloudless setup.

**Comment by terzo33** [comment]
> Can someone try [feat: Implemented response to V5 HEARTBEAT](https://github.com/davidrapan/ha-solarman/commit/1f148857e670ac190ee8a0f52c480974197be665)?
> 
> It should solve time synchronization for cloudless setup.

@davidrapan how can I do?

**Comment by davidrapan** [comment]
Just run the latest code.

`Meaning you have to manually update the code as it's not part of any release yet.`

**Comment by davidrapan** [comment]
@terzo33, I had to do some fixes already [fix: Sequence number for response to V5 HEARTBEAT](https://github.com/davidrapan/ha-solarman/commit/7b2645983c8ba329581435867f8ef93526f571cd) 😉

**Comment by davidrapan** [comment]
@terzo33, I had to do some fixes already [fix: Sequence number for response to V5 HEARTBEAT](https://github.com/davidrapan/ha-solarman/commit/d7c4e76a829aacdafa588002d707614325e41984) 😉

**Comment by terzo33** [comment]
Done 💪

**Comment by davidrapan** [comment]
One more [fix: Sequence number for response to V5 HEARTBEAT](https://github.com/davidrapan/ha-solarman/commit/d7c4e76a829aacdafa588002d707614325e41984) 😉😆

**Comment by davidrapan** [comment]
One more [fix: Run response to HEARTBEAT in asyncio.ensure_future](https://github.com/davidrapan/ha-solarman/commit/dd7aa490a92ab09c9edba7b9fa0d776acec627a2) 😉😆

**Comment by davidrapan** [comment]
One more [fix: Run response to HEARTBEAT in asyncio.ensure_future](https://github.com/davidrapan/ha-solarman/commit/146b5284d9d690d0a99f2d37233382975f1f1905) 😉😆

**Comment by terzo33** [comment]
Today it's a rainy day... i can't try :(

**Comment by terzo33** [comment]
> One more [fix: Run response to HEARTBEAT in asyncio.ensure_future](https://github.com/davidrapan/ha-solarman/commit/146b5284d9d690d0a99f2d37233382975f1f1905) 😉😆

@davidrapan it doesn't seem to work

![Screenshot_20241219_101736_Samsung Internet.jpg](https://github.com/user-attachments/assets/9e562ee0-4ca4-4ee8-9271-dccfd9f44b1b)

**Comment by davidrapan** [comment]
The changes purpose is not to anyhow change update period of those entities but to enable time synchronization in cloudless environment!

`I just used this thread to find someone with microinverter to test the functionality.`

**Comment by davidrapan** [comment]
> yes but it doesn't seem to work because the data is updated every 5 minutes as it happens via cloud. is there anything I can do to help you?

You do not understand. It has nothing to do with this.

**Comment by davidrapan** [comment]
@terzo33, show me please your debug log (hope you recorded some or still have chance to).

**Comment by terzo33** [comment]
> @terzo33, show me please your debug log (hope you recorded some or still have chance to).

@davidrapan ![home-assistant_solarman_2024-12-20T13-14-02.228Z.txt](https://github.com/user-attachments/files/18211102/home-assistant_solarman_2024-12-20T13-14-02.228Z.txt)

**Comment by davidrapan** [comment]
Looks like it does work:
2024-12-20 09:34:06.435 DEBUG (MainThread) [custom_components.solarman.api] [38????????] V5_HEARTBEAT: a5 01 00 10 47 02 d7 cb ea 91 e6 00 5d 15
2024-12-20 09:34:06.436 DEBUG (MainThread) [custom_components.solarman.api] [38????????] V5_HEARTBEAT RESPONSE: a5 0a 00 10 17 03 cb ea 91 e6 00 01 fe 2b 65 67 00 00 00 00 56 15

**Comment by terzo33** [comment]
> Looks like it does work:
> ```
> 2024-12-20 09:34:06.435 DEBUG (MainThread) [custom_components.solarman.api] [38????????] V5_HEARTBEAT: a5 01 00 10 47 02 d7 cb ea 91 e6 00 5d 15
> 2024-12-20 09:34:06.436 DEBUG (MainThread) [custom_components.solarman.api] [38????????] V5_HEARTBEAT RESPONSE: a5 0a 00 10 17 03 cb ea 91 e6 00 01 fe 2b 65 67 00 00 00 00 56 15
> ```

@davidrapan great, but what purpose will this thing you're doing serve? What improvement will it bring?

**Comment by davidrapan** [comment]
I told you, time synchronization for installations in a cloudless (disconnected from solarman servers) environment.

**Comment by terzo33** [comment]
> I told you, time synchronization for installations in a cloudless (disconnected from solarman servers) environment.

yes I understood this, but by disconnecting from the solarman cloud can we also have improvements on the data or extrapolate more data?

**Comment by davidrapan** [comment]
It should prevent timeouts but this functionality has nothing to do w/ subject of this thread. I already said it, that I just used this thread to announce this feature. 😉

**Comment by msaphoto06** [comment]
Tjs pas de solution a ce problème ?

**Comment by maybetaken** [comment]
Hi,
I have the same question with M80G4, I live in China and I am trying to reach the technician for help. Once there are any feedbacks, I will update here.

**Comment by davidrapan** [comment]
You're not the only one who didn't get an answer. 😉

**Comment by amiko24** [comment]
Good evening,
I apologize, but I have to ask the same question. I use the Solarman integration in version 25.08.16. The update time for MI inverters is five minutes. Is there a way to shorten the update interval? I tried to set the update interval for the PV group to 5 seconds in the YAML file, even though the default update interval is already set to 5 seconds. Unfortunately, it didn't work.

![Image](https://github.com/user-attachments/assets/20605091-f6cb-451a-acf5-fe4fd554272f)
![Image](https://github.com/user-attachments/assets/ad071ff4-c1e6-4712-b516-8d9d0b08200f)
![Image](https://github.com/user-attachments/assets/200ab4b6-2e19-4281-86dd-11954f183b55)

**Comment by CrazyUs3r** [comment]
Hey @amiko24 this not a problem from solarman your in the card "5-Minuten-Aggregation" all 5 min filter hy ha

**Comment by amiko24** [comment]
Is that really the case? Solarman's requests are independent of the Wi-Fi logger. When I send a command to the inverter via the Deye cloud, I receive a response without having to wait five minutes.

**Comment by Adminius** [comment]
> Is that really the case? Solarman's requests are independent of the Wi-Fi logger. When I send a command to the inverter via the Deye cloud, I receive a response without having to wait five minutes.

Yeah, but up to 5 min old data, so mostly outdated in cloudy conditions

**Comment by amiko24** [comment]
How can you shorten the update interval? Send an email to Deye Support and ask them to set the update interval to one minute? Can I do it myself?

**Comment by Adminius** [comment]
I tried via support, and they changed frome 5 min to 1 min. But some time later it went back to 5 min again. And this change apply only for online data and not local data.
I now use AC energy meter for my MIs to get 1s data and this integration to get kWh only :/

**Comment by amiko24** [comment]
OK, thank you

**Comment by davidrapan** [comment]
This is the problem w/ microinverters. You can try to retrieve data w/ any interval you want, but the device will always respond w/ the same old data w/ an update interval of 5 min (in some cases 1 min).
